test/Microsoft.AspNetCore.Server.KestrelTests/FrameResponseHeadersTests.cs

```diff
@@ -243,5 +257,27 @@ public void ContentLengthValueClearedWhenHeadersCleared()
 
             Assert.Equal(null, headers.HeaderContentLengthValue);
         }
+
+        private static long ParseLong(string value)
+        {
+            return long.Parse(value, NumberStyles.AllowLeadingWhite | NumberStyles.AllowTrailingWhite, CultureInfo.InvariantCulture);
+        }
+
+        public static TheoryData<string> GoodContentLengths => new TheoryData<string>
+        {
+            "0",
+            "00",
+            "042",
+            "42",
+        };
+
+        public static TheoryData<string> BadContentLengths => new TheoryData<string>
+        {
+            "",
```

Comment:
"ü" will be rejected by `ValidateHeaderCharacters()`. But yeah I'll add a few more cases.